dora/underfs/hdfs/src/test/java/alluxio/underfs/hdfs/hdfs3/HdfsUnderFileSystemIntegrationTest.java

```diff
@@ -30,4 +34,20 @@ public void testWriteMultiBlockFileTest() throws Exception {
   public void testException() throws Exception {
     hdfsDownDuringUploadTest();
   }
+
+  @Test
+  public void testSetAndGetXAttribute() throws Exception {
+    // create empty file
+    String testFilePath = "/empty_file";
+    OutputStream os = mUfs.create(testFilePath, getCreateOption());
+    os.close();
+    assertEquals(0, mUfs.getStatus(testFilePath).asUfsFileStatus().getContentLength());
+
+    // setXAttr
+    String attrKey = "key1";
+    String attrValue = "value1";
+    mUfs.setAttribute(testFilePath, attrKey, attrValue.getBytes());
+    Map<String, String> attrMap =  mUfs.getAttribute(testFilePath);
+    assertEquals(attrMap.get(attrKey), attrValue);
```

Comment:
i figure some other UT cases can be added on corner cases at least, like unsetting an attribute, setting attr twice and observe a successful overwrite etc.